gax/src/main/java/com/google/api/gax/batching/v2/BatchingCallSettings.java

```diff
@@ -0,0 +1,134 @@
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.batching.v2;
+
+import com.google.api.core.BetaApi;
+import com.google.api.gax.retrying.RetrySettings;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.api.gax.rpc.UnaryCallSettings;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.common.base.Preconditions;
+import java.util.Set;
+
+/**
+ * A settings class to configure a {@link UnaryCallable} for calls to an API method that supports
+ * batching. The settings are provided using an instance of {@link BatchingSettings}.
+ */
+@BetaApi("The surface for batching is not stable yet and may change in the future.")
+public final class BatchingCallSettings<ElementT, ElementResultT, RequestT, ResponseT>
+    extends UnaryCallSettings<RequestT, ResponseT> {
+  private final BatchingDescriptor<ElementT, ElementResultT, RequestT, ResponseT>
+      batchingDescriptor;
+  private final BatchingSettings batchingSettings;
+
+  public BatchingDescriptor<ElementT, ElementResultT, RequestT, ResponseT> getBatchingDescriptor() {
+    return batchingDescriptor;
+  }
+
+  public BatchingSettings getBatchingSettings() {
+    return batchingSettings;
+  }
+
+  private BatchingCallSettings(Builder<ElementT, ElementResultT, RequestT, ResponseT> builder) {
+    super(builder);
+    this.batchingDescriptor = builder.batchingDescriptor;
+    this.batchingSettings = Preconditions.checkNotNull(builder.batchingSettings);
+  }
+
+  public static <ElementT, ElementResultT, RequestT, ResponseT>
```

Comment:
Lets mark this as InternalOnly("for google-cloud-java use only") and add javadoc directing end users to use toBuilder on the existing instance exposed by clients?

Reply:
Added the Javadoc to refer to `BatchingCallSettings#toBuilder()`. Please have a look.